src/main/kotlin/io/hsar/wh40k/combatsimulator/logic/TacticalActionStrategy.kt

```diff
@@ -2,34 +2,68 @@ package io.hsar.wh40k.combatsimulator.logic
 import io.hsar.wh40k.combatsimulator.random.AverageDice
 
 object TacticalActionStrategy : ActionStrategy {
 
-    override fun decideTurnActions(world: World, thisUnit: UnitInstance, possibleActions: Collection<TurnAction>): List<TurnAction> {
+    override fun decideTurnActions(world: World, thisUnit: UnitInstance, possibleActionOptions: Collection<ActionOption>): List<TurnAction> {
         // Shit implementation - units will only ever aim and fire their max damage attack
-        val aimAction = possibleActions
+        val aimAction = possibleActionOptions
                 .find { action ->
-                    action is TurnAction.HalfAim
+                    action is ActionOption.HalfAim
+                }
+                ?.let { action ->
+                    AimAction(action)
                 }
 
-        val maxDamageAttackAction = possibleActions
+        val maxDamageAttackAction = possibleActionOptions
                 .filter { action ->
                     action is DamageCausingAction
                 }
                 .map { eachDamageCausingAction ->
+                    (eachDamageCausingAction as DamageCausingAction)
+                }
+                .let { damageCausingActions ->
+                    performTargeting(world, thisUnit, damageCausingActions)
+                    // this will produce a list of physically possible targeted actions
+                }
+                .map { eachTargetedAction ->
                     // Associate each action with an estimate of its damage
-                    eachDamageCausingAction to (eachDamageCausingAction as DamageCausingAction).damage
-                            .let { damageString ->
-                                AverageDice.roll(damageString)
-                            }
+                    when (eachTargetedAction.action) {
+                        is DamageCausingAction -> eachTargetedAction.action.damage
+                                .let { damageString ->
+                                    eachTargetedAction to AverageDice.roll(damageString)
+                                }
+                        else -> eachTargetedAction to 0
+                    }
                 }
                 .maxBy { (_, expectedDamage) -> expectedDamage }
                 ?.first
 
-
         return listOfNotNull(aimAction, maxDamageAttackAction)
     }
 
-    //TODO add collision detection
+    fun performTargeting(world: World, thisUnit: UnitInstance, possibleAttacks: List<DamageCausingAction>): List<TargetedAction> {
+
+        // This will create a list containing a TargetedAction for each feasible action/target pair
+        return possibleAttacks.map { possibleAttack ->
+            world.getAdversaries(thisUnit).map { adversary ->
+                TargetedAction(possibleAttack as ActionOption, adversary)
+            }
+
+        }.flatten().filter { targetedAction -> // filter out options that are not possible due to range etc
```

Comment:
[Minor] Stylistically I prefer to space things out so that there's a maximum of one thing happening per line, i.e.
```
.flatten()
.filter { ...
```

Reply:
Implemented